File: JS/Tema1_Ejercicio2.js
/*
 * TEMA 1: INTRODUCCIÓN BÁSICA - EJERCICIO 2
 * ===========================================
 * Programa que clasifica un número en positivo, negativo o cero.
 * 
 * EXPLICACIÓN:
 * - Operador ternario (? :)
 * - Sentencias condicionales: if-else
 * - Template literals para formatear strings
 */

const numero = 42;

// Clasificar número
let clasificacion;
if (numero > 0) {
    clasificacion = "Positivo";
} else if (numero < 0) {
    clasificacion = "Negativo";
} else {
    clasificacion = "Cero";
}

console.log(`=== CLASIFICACIÓN DE NÚMEROS ===`);
console.log(`El número ${numero} es: ${clasificacion}`);

// Usar operador ternario para el mismo resultado
const tipo = numero > 0 ? "Positivo" : (numero < 0 ? "Negativo" : "Cero");
console.log(`Con ternario: ${tipo}`);

/*
 * SALIDA ESPERADA:
 * === CLASIFICACIÓN DE NÚMEROS ===
 * El número 42 es: Positivo
 * Con ternario: Positivo
 */

File: Java/Tema1_Ejercicio2.java
/*
 * TEMA 1: INTRODUCCIÓN BÁSICA - EJERCICIO 2
 * ==========================================
 * Escriba un programa que lea un número entero y muestre si es positivo, 
 * negativo o cero. Además, muestre el valor absoluto.
 * 
 * EXPLICACIÓN:
 * - Usamos condicionales para validar el número
 * - Math.abs() nos da el valor absoluto de un número
 */

import java.util.Scanner;

public class Tema1_Ejercicio2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite un número: ");
        int numero = scanner.nextInt();
        
        // Determinar el tipo de número
        String tipo;
        if (numero > 0) {
            tipo = "positivo";
        } else if (numero < 0) {
            tipo = "negativo";
        } else {
            tipo = "cero";
        }
        
        int valorAbsoluto = Math.abs(numero);
        
        System.out.println("El número es: " + tipo);
        System.out.println("Valor absoluto: " + valorAbsoluto);
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * ===============
 * Digite un número: -15
 * El número es: negativo
 * Valor absoluto: 15
 */
